src/main/java/nextstep/member/application/MemberAuthenticationService.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.member.application;
+
+import nextstep.authentication.application.AuthenticationService;
+import nextstep.authentication.domain.AuthenticationInformation;
+import nextstep.authentication.application.dto.GithubProfileResponse;
+import nextstep.authentication.domain.LoginMember;
+import nextstep.member.domain.Member;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional(readOnly = true)
+@Service
+public class MemberAuthenticationService implements AuthenticationService {
+
+    private final MemberService memberService;
+
+    public MemberAuthenticationService(MemberService memberService) {
+        this.memberService = memberService;
+    }
+
+    @Override
+    public AuthenticationInformation findMemberByEmail(String email) {
+        Member member = memberService.findMemberByEmail(email);
+        return new AuthenticationInformation(member.getEmail(), member.getId(), member.getPassword());
+    }
```

Comment:
```suggestion
@Service
public class MemberAuthenticationService implements AuthenticationService {

    private final MemberService memberService;

    public MemberAuthenticationService(MemberService memberService) {
        this.memberService = memberService;
    }

    @Override
    @Transactional(readOnly = true)
    public AuthenticationInformation findMemberByEmail(String email) {
        Member member = memberService.findMemberByEmail(email);
        return new AuthenticationInformation(member.getEmail(), member.getId(), member.getPassword());
    }
```

저는 하나의 클래스에서 `@Transcational`이 혼합됐을 때 각각 선언해주고 있긴합니다.

Reply:
저는 클래스에 붙여주면 중복을 피할 수 있다고 생각했는데, 리뷰어님은 각각 붙이시는 이유가 있을까요?

Reply:
아! [트랜잭션리뷰](https://github.com/next-step/atdd-subway-favorite/pull/699#discussion_r1741258012) 이 리뷰와 연결되겠네요!!

제가 의도한 부분만 DB 커넥션을 열어주어 DB pool이 낭비되는 상황을 줄이고 동시에 connection이 지연되지 않도록 할 수 있겠군요!!